cmd/metal-api/internal/grpc/supwdServer.go

```diff
@@ -0,0 +1,22 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	v1 "github.com/metal-stack/metal-api/pkg/api/v1"
+	"io/ioutil"
+	"strings"
+)
+
+func (s *Server) FetchSupermetalPassword(ctx context.Context, req *v1.SupermetalPasswordRequest) (*v1.SupermetalPasswordResponse, error) {
+	defer ctx.Done()
+
+	bb, err := ioutil.ReadFile(fmt.Sprintf("/supwds/%s.supwd", req.PartitionID))
```

Comment:
As HMACs are currently also just set as environment variable, I would do it the same way for now here.

We can create a follow-up issue where we move things like passwords to configuration files, such that a user cannot see the content from watching the pod spec only.

Reply:
I've already prepared a secret and a corresponding /supermetal mountpoint in metal-roles, which is pretty much the same mechanism we already have regarding the grpc certificates.
That being said I would leave this here as is, since it's also the final solution.

Reply:
But then we would also need the HMACs to be put into separate files when we want everything to be uniform. This means, that in the end there would be six more files where we do `ioutil.ReadFile`, which I think is not as safe as the existing initialization because we need to ensure that all required parameters are present on startup, otherwise don't even start the API.

Also, if this password can be different for every partition... isn't this actually part of the partition configuration?

Reply:
I agree.
Another solution would be to use one bmc admin password for all machines of all vendors.
For this we could leave this implementation as is despite changing the path to something like this `/bmc/admin.pwd`

Reply:
I think, if it is just a single file and the existence is ensured at startup, I am fine. If it would be optional (non-breaking for deployments) this would be even better. Only set a password when a user explicitly provides one, otherwise just do not do anything.

Reply:
Since connecting to a machine's bmc is already protected by our mgmt-net I'm also fine using only one password that rules them all.
If we make this password optional, we do not fix the issue of loosing machines (at least from remote access), which happened in the past, if only rarely. Additionally we would have to go the second path, namely to disable the bmc admin user with its publicly-known password and create an new one (`supermetal`) with admin privileges.

Anyways, this second path would have the benefit that an admin user do not care about the machine's vendor in order to access it's bmc, since the admin usernames differ from vendor to vendor.